contracts/modules/claims/Incidents.sol

```diff
@@ -0,0 +1,234 @@
+/* Copyright (C) 2021 NexusMutual.io
+
+  This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+    along with this program.  If not, see http://www.gnu.org/licenses/ */
+
+pragma solidity ^0.5.0;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "../../abstract/MasterAware.sol";
+import "../../interfaces/IPooledStaking.sol";
+import "../capital/Pool.sol";
+import "../claims/ClaimsData.sol";
+import "../claims/ClaimsReward.sol";
+import "../cover/QuotationData.sol";
+import "../governance/MemberRoles.sol";
+import "../token/TokenController.sol";
+
+contract Incidents is MasterAware {
+  using SafeERC20 for IERC20;
+  using SafeMath for uint;
+
+  struct Incident {
+    uint32 date;
+    uint112 priceBefore;
+    uint112 priceAfter;
+    address coveredToken;
+  }
+
+  // contract identifiers
+  enum ID {CD, CR, QD, TC, MR, P1, PS}
+
+  mapping(uint => address payable) contracts;
+
+  Incident[] public incidents;
+
+  // covered token => peg token
+  mapping(address => address) public underlyingTokens;
```

Comment:
what's the plan for migrating these once we get more than 1 supported underlying token? since this is a proxy contract memory layout will stay the same, we should prepare this multiple underlying tokens